new-log-viewer/src/components/Layout.tsx

```diff
@@ -17,31 +16,42 @@ import {
 const Layout = () => {
     const {
         logData,
-        loadFile,
-        logEventNum,
+        pageNum,
+        numEvents,
     } = useContext(StateContext);
+    const {logEventNum} = useContext(UrlContext);
+
+    const handleLogEventNumInputChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
+        updateWindowHashParams({logEventNum: Number(ev.target.value)});
+    };
 
-    useEffect(() => {
-        const urlSearchParams = new URLSearchParams(window.location.search);
-        const filePath = urlSearchParams.get("filePath");
-        if (null !== filePath) {
-            loadFile(filePath);
-        }
-    }, [loadFile]);
+    const handleCopyLinkButtonClick = () => {
+        copyToClipboard({}, {logEventNum: numEvents});
+    };
 
     return (
         <>
             <div>
                 <h3>
                     LogEventNum -
                     {" "}
-                    {logEventNum}
+                    <input
+                        type={"number"}
+                        value={null === logEventNum ?
+                            -1 :
```

Comment:
-1 just to test the clamping logic?

Reply:
Let's change this into `1` since we may want to reserve `-1` for the last log event.